op_crates/fetch/21_blob.js

```diff
@@ -0,0 +1,289 @@
+     * @returns {Uint8Array}
+     */
+  function processBlobParts(parts, endings) {
+    /** @type {Uint8Array[]} */
+    const bytesArrays = [];
+    for (const element of parts) {
+      if (element instanceof ArrayBuffer) {
+        bytesArrays.push(new Uint8Array(element.slice(0)));
+      } else if (ArrayBuffer.isView(element)) {
+        const buffer = element.buffer.slice(
+          element.byteOffset,
+          element.byteOffset + element.byteLength,
+        );
+        bytesArrays.push(new Uint8Array(buffer));
+      } else if (element instanceof Blob) {
+        bytesArrays.push(
+          new Uint8Array(element[_byteSequence].buffer.slice(0)),
+        );
+      } else if (typeof element == "string") {
+        let s = element;
+        if (endings == "native") {
+          s = convertLineEndingsToNative(s);
+        }
+        bytesArrays.push(utf8Encoder.encode(s));
+      } else {
+        throw new TypeError("Unreachable code (invalild element type)");
+      }
+    }
+    return concatUint8Arrays(...bytesArrays);
+  }
+
+  /**
+   * @param {string} str 
+   * @returns {string}
+   */
+  function normalizeType(str) {
+    let normalizedType = str;
+    if (!/^[\x20-\x7E]*$/.test(str)) {
+      normalizedType = "";
+    }
+    return normalizedType.toLowerCase();
+  }
+
+  const _byteSequence = Symbol("[[ByteSequence]]");
+
+  class Blob {
+    /** @type {string} */
+    #type;
+
+    /** 
+     * @param {BlobPart[]} [blobParts]
+     * @param {BlobPropertyBag} [options]
+     */
+    constructor(blobParts, options) {
+      if (blobParts === undefined) {
+        blobParts = [];
+      }
+      if (typeof blobParts !== "object") {
+        throw new TypeError(
+          `Failed to construct 'Blob'. blobParts cannot be converted to a sequence.`,
+        );
+      }
+
+      const parts = [];
+      const iterator = blobParts[Symbol.iterator]?.();
+      if (iterator === undefined) {
+        throw new TypeError(
+          "Failed to construct 'Blob'. The provided value cannot be converted to a sequence",
+        );
+      }
+      while (true) {
+        const { value: element, done } = iterator.next();
+        if (done) break;
+        if (
+          ArrayBuffer.isView(element) || element instanceof ArrayBuffer ||
+          element instanceof Blob
+        ) {
+          parts.push(element);
+        } else {
+          parts.push(String(element));
+        }
+      }
+
+      if (!options || typeof options == "function") {
+        options = {};
+      }
+      if (typeof options != "object") {
+        throw new TypeError(
+          `Failed to construct 'Blob'. options is not an object.`,
+        );
+      }
+      const endings = options?.endings?.toString() ?? "transparent";
+      const type = options?.type?.toString() ?? "";
+
+      /** @type {Uint8Array} */
+      const bytes = processBlobParts(parts, endings);
+      this[_byteSequence] = bytes;
+      this.#type = normalizeType(type);
```

Comment:
I personally have avoided private members in spec following code, even if they aren't accessed out of their class, for consistency sake.  If the spec calls for it to be an internal slot, maybe just make it a symbol like `_byteSequence`?

Reply:
Here the spec doesn't actually call for it to be an internal slot. We only make it a private type because it is a readonly property (it is read back to the user via `get type`).

Reply:
👍